R/valley.R

```diff
@@ -0,0 +1,273 @@
+#'Retrieve asset urls for the intersection of a bounding box with a 
+#'remote STAC endpoint 
+#' 
+#' @param endpoint url of (remote) STAC endpoint
+#' @param collection STAC collection to be queried
+#' @param bb A bounding box (compliant with CRiSp, i.e. as a matrix with 4 elements: xmin, ymin, xmax, ymax)
+#'
+#' @return A list of urls for the assets in the collection overlapping with the specified bounding box
+#' @export
+get_stac_asset_urls <- function(bb, endpoint="https://earth-search.aws.element84.com/v1", collection="cop-dem-glo-30", limit=100){
+    s_obj <- stac(endpoint)
+    it_obj <- s_obj |>
+    stac_search(collections = collection,
+              bbox = as.vector(bb)) |>
+    get_request()
+    asset_urls <- rstac::assets_url(it_obj)
+}
+
+#' retrieve STAC records (of a DEM) corresponding to a list of asset urls,
+#' crop and merge with a specified bounding box to create a dem of the 
+#' specified region
+#' 
+#' @param asset_urls a list of STAC records to be retrieved
+#' @param bb A bounding box (compliant with CRiSp, i.e. as a matrix with 4 elements: xmin, ymin, xmax, ymax)
+#' 
+#' @return A a merged dem fromm retrieved assets cropped to the bounding box
+#' @export
+load_raster <- function(raster_urlpaths, bb){
+    raster_urlpaths |>
+    lapply(rast) |>
+    lapply(crop, as.vector(t(bb))) |>
+    do.call(merge, args=_)
+}
+
+
+#'Load dem either from file or a STAC endpoint
+#' 
+#' @param resource from whcih to source dem. Can be "STAC" or "FILE".
+#' if "STAC" the parameters the following parameters must/can be supplied as named parameters
+#' @param endpoint STAC endpoint to use (required)
+#' @param collection STAC collection to use (required)
+#' @param limit limit of records to retrieve (optional)
+#' 
+#' @return dem 
+#' @export
+get_dem <- function(bb, resource="STAC",...){
+    kwargs <- list(...)
+    if (resource == "STAC"){
+        if(length(kwargs) && !is.null(...)){
+            endpoint = kwargs$endpoint
+            collection = kwargs$collection
+            asset_urls <- get_stac_asset_urls(bb,endpoint=endpoint,collection=collection
+        } else {
+            asset_urls <- get_stac_asset_urls(bb)
+        }
+        dem <- get_stac_dem(bb, asset_urls) 
```

Comment:
I cannot find `get_stac_dem()`. Shouldn't this be `load_raster()`?

Reply:
yes, this has already been changed